Ryujinx.Ava/UI/Models/InMemoryLogTarget.cs

```diff
@@ -0,0 +1,78 @@
+using Avalonia.Media;
+using Ryujinx.Common.Logging;
+using System.Collections.ObjectModel;
+
+namespace Ryujinx.Ava.UI.Models
+{
+    public class InMemoryLogTarget : ILogTarget
+    {
+        public class Entry
+        {
+            public Color Color { get; set; }
+            public string Text { get; set; }
+        }
+
+        private static Color GetLogColor(LogLevel level) => level switch {
+            LogLevel.Info    => Colors.White,
+            LogLevel.Warning => Colors.Yellow,
+            LogLevel.Error   => Colors.Red,
+            LogLevel.Stub    => Colors.DarkGray,
+            LogLevel.Notice  => Colors.Cyan,
+            LogLevel.Trace   => Colors.DarkCyan,
+            LogLevel.StdErr  => Colors.Gray,
+            _                => Colors.Gray,
+        };
+
+        private static readonly InMemoryLogTarget _instance = new InMemoryLogTarget("inMemory");
+
+        private readonly ILogFormatter _formatter;
+        private readonly string _name;
+
+        private const int _maximumSize = 20000;
+
+        public ObservableCollection<Entry> Entries;
+
+        string ILogTarget.Name { get => _name; }
```

Comment:
```suggestion
        string ILogTarget.Name => _name;
```
I think this is valid too?